aws/resource_aws_location_tracker.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/locationservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsLocationTracker() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLocationTrackerCreate,
+		Read:   resourceAwsLocationTrackerRead,
+		Delete: resourceAwsLocationTrackerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
add validation for len and regex as mentioned at https://docs.aws.amazon.com/location-trackers/latest/APIReference/API_CreateTracker.html#API_CreateTracker_RequestSyntax